### PR TITLE
Add NaN validation to Evaluator

### PR DESCRIPTION
**Pull request by abdulfatir**
*Issue #, if available:* #2560

*Description of changes:*
This PR adds NaN validation to the evaluator via a `strict` parameter which defaults to True. When `strict` is `True`, an error is raised if the forecasts contain NaN values, otherwise a warning is logged. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by jaheba** [comment]
As I said in #2560, I'm against checking this in the evaluator.

File: src/gluonts/evaluation/_base.py [hunk 1]
```diff
@@ -150,11 +182,14 @@ class Evaluator:
     ignore_invalid_values
         Ignore `NaN` and `inf` values in the timeseries when calculating
         metrics.
-    aggregation_strategy:
+    aggregation_strategy
         Function for aggregating per timeseries metrics.
         Available options are:
         aggregate_valid | aggregate_all | aggregate_no_nan
         The default function is aggregate_no_nan.
+    strict
```
Comment (jaheba) [inline src/gluonts/evaluation/_base.py]: Maybe something like `allow_nan_forecast` is more explicit?

Comment (abdulfatir) [reply src/gluonts/evaluation/_base.py]: Changed.

**Comment by lostella** [comment]
image

**Comment by lostella** [comment]
In all seriousness: I believe this is the desired behaviour, and the previous one was incorrect in that numbers coming out of evaluation can be misleading if a model happens to output a `nan`.

**Review by lostella — APPROVED** [review]

**Pull request merged**